Code/GraphMol/RGroupDecomposition/RGroupData.h

```diff
@@ -59,7 +59,10 @@ struct RGroupData {
               std::inserter(attachments, attachments.end()));
 
     mols.push_back(newMol);
-    smilesVect.emplace_back(MolToSmiles(*newMol, true));
+    static const std::regex remove_isotopes_regex("\\[\\d*\\*\\]");
+    std::string smiles_no_isotopes = std::regex_replace(
+        MolToSmiles(*newMol, true), remove_isotopes_regex, "*");
+    smilesVect.push_back(std::move(smiles_no_isotopes));
```

Comment:
For some reason, I just don't like the regular expression usage, but I'm going to ignore my feelings for now and profile the change.

Let's make it easier on copy elision and make this:

```
smilesVect.push_back(
  std::regex_replace(MolToSmiles(*newMol, true), remove_isotopes_regex, "*")
);
```

Reply:
By profile the change I mean see if this is faster than removing the isotope from the atom before making the smiles string :)

Reply:
I broke it in two just for clarity - with the `std::move` the two constructs should be equivalent once compiled. Anyway, I am happy to do the change.
I don't particularly fancy the regex either, it just seems the simplest way to get job done.

Reply:
And I'm glad we came to the same conclusions, if the scoring function couldn't align these, there was something seriously wrong there.

This probably fixes some edge cases with multiple core replacements as well :)

Reply:
Done.

Reply:
> And I'm glad we came to the same conclusions, if the scoring function couldn't align these, there was something seriously wrong there.

Absolutely! Glad that you insisted as it prompted me to reconsider my initial patch.